cbor/src/main/java/com/fasterxml/jackson/dataformat/cbor/CBORParser.java

```diff
@@ -46,17 +48,92 @@ public static int collectDefaults()
             }
             return flags;
         }
-        
+
         private Feature(boolean defaultState) {
             _defaultState = defaultState;
             _mask = (1 << ordinal());
         }
-        
+
         @Override public boolean enabledByDefault() { return _defaultState; }
         @Override public int getMask() { return _mask; }
         @Override public boolean enabledIn(int flags) { return (flags & _mask) != 0; }
     }
 
+    /**
+     * Class for keeping track of tags in an optimized manner.
+     *
+     * @since 2.15
+     */
+    public static final class TagList
+    {
+        public TagList() {
+            _tags = new int[8];
+            _tagCount = 0;
+        }
+
+        /**
+         * Checks whether the tag list is empty.
+         *
+         * @return {@code true} if there are no tags, {@code false} if there are tags..
+         */
+        public boolean isEmpty() {
+            return _tagCount == 0;
+        }
+
+        /**
+         * Clears the tags from the list.
+         */
+        public void clear() {
+            _tagCount = 0;
+        }
+
+        /**
+         * Adds a tag to the list.
+         *
+         * @param tag The tag to add.
+         */
+        public void add(int tag) {
+            if (_tagCount == _tags.length) {
+                // Linear growth since we expect a small number of tags.
+                int[] newTags = new int[_tagCount + 8];
+                System.arraycopy(_tags, 0, newTags, 0, _tagCount);
+                _tags = newTags;
+            }
+
+            _tags[_tagCount++] = tag;
+        }
+
+        /**
+         * Checks if a tag is present.
+         *
+         * @param tag The tag to check.
+         * @return {@code true} if the tag is present, {@code false} if it is not.
+         */
+        public boolean contains(int tag) {
+            for (int i = 0; i < _tagCount; ++i) {
+                if (_tags[i] == tag) {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        /**
+         * Gets the first tag in the list. This is primarily to support the legacy API.
+         *
+         * @return The first tag or -1 if there are no tags.
+         */
+        public int getFirstTag() {
+            if (_tagCount == 0) {
+                return -1;
+            }
+            return _tags[0];
+        }
+
+        private int[] _tags;
```

Comment:
Maybe add `size()` method? (`return _tagCount;`)

Reply:
I've added the method as requested.